src/deps/c_ares.zig

```diff
@@ -372,11 +463,92 @@ pub const struct_ares_caa_reply = extern struct {
     length: usize,
 };
 pub const struct_ares_srv_reply = extern struct {
-    next: [*c]struct_ares_srv_reply,
+    next: ?*struct_ares_srv_reply,
     host: [*c]u8,
     priority: c_ushort,
     weight: c_ushort,
     port: c_ushort,
+    const JSC = bun.JSC;
+
+    pub fn toJSArray(this: *struct_ares_srv_reply, parent_allocator: std.mem.Allocator, globalThis: *JSC.JSGlobalObject) JSC.JSValue {
+        var stack = std.heap.stackFallback(2048, parent_allocator);
+        var arena = std.heap.ArenaAllocator.init(stack.get());
+        defer arena.deinit();
+
+        var allocator = arena.allocator();
+        var count: usize = 0;
+        var srv: ?*struct_ares_srv_reply = this;
+        while (srv != null) : (srv = srv.?.next) {
+            count += 1;
+        }
+
+        const array = JSC.JSValue.createEmptyArray(globalThis, count);
+
+        srv = this;
+        var i: u32 = 0;
+        while (srv != null) {
+            var node = srv.?;
+            array.putIndex(globalThis, i, node.toJS(globalThis, allocator));
+            srv = node.next;
+            i += 1;
+        }
+
+        return array;
+    }
+
+    pub fn toJS(this: *struct_ares_srv_reply, globalThis: *JSC.JSGlobalObject, allocator: std.mem.Allocator) JSC.JSValue {
+        var obj = JSC.JSValue.createEmptyObject(globalThis, 4);
+        // {
+        //   priority: 10,
+        //   weight: 5,
+        //   port: 21223,
+        //   name: 'service.example.com'
+        // }
+
+        obj.put(globalThis, JSC.ZigString.static("priority"), JSC.JSValue.jsNumber(this.weight));
+        obj.put(globalThis, JSC.ZigString.static("weight"), JSC.JSValue.jsNumber(this.priority));
+        obj.put(globalThis, JSC.ZigString.static("port"), JSC.JSValue.jsNumber(this.port));
+
+
+
+        const len = std.mem.len(this.host);
+        var host = this.host[0..len];
+        var name = JSC.ZigString.fromUTF8(host).clone(allocator) catch unreachable;
```

Comment:
no need to clone here, `toValueGC` (below) will clone it

cloning here will leak the memory

Reply:
done